lib/conversation/replies.js

```diff
@@ -40,22 +40,23 @@ function customMsg(fn) {
  * @return {function}    wrapped function
  */
 function renderEndConversationMsg(fn) {
-  return (replyTemplate, args) => helpers.renderMessageTemplate(args.req, replyTemplate)
-    .then((replyText) => {
-      if (!args.req.user) {
-        const signup = args.req.signup;
-        const userId = args.req.userId;
-        return helpers.sendResponseForSignup(args.res, signup, userId, replyText, replyTemplate);
-      }
-
-      args.replyText = replyText;
-      fn(args);
-      stathat.postStat(`campaignbot:${replyTemplate}`);
-      BotRequest.log(args.req, 'campaignbot', replyTemplate, replyText);
-
-      return helpers.sendResponse(args.res, 200, replyText, replyTemplate);
-    })
-    .catch(err => helpers.sendErrorResponse(args.res, err));
+  return (replyTemplate, args) => {
+    if (!args.req.user) {
+      const signup = args.req.signup;
+      return helpers.sendResponseForSignup(args.res, signup, replyTemplate);
```

Comment:
I know this is something it's happening throughout the codebase (wasn't introduced by this commit) and we might have to do cleanup at a later point, but if a function returns a promise at any point, we should promisify all branches of the function, just so we don't end up returning an object to a function that was expecting a promise under some circumstances or not rejecting when that is what we intended to do. I fixed something like that in the code I am working on in [Conversations](https://github.com/DoSomething/gambit-conversations/blob/master/lib/helpers.js#L276) (fix not pushed yet).

Reply:
Agreed - this `/lib/conversation` will definitely need some major cleanup once we go live -- this was a workaround to avoid making changes to function calls in all of the various middleware.